src/abstractdataframe/show.jl

```diff
@@ -410,21 +404,27 @@ function showrows(io::IO,
     colwidths = maxwidths .+ textwidth(" ") .+ textwidth(" |")  # padding+border
     colwidths[end] += textwidth("|")  # add row column initial border
     coltextwidths = colwidths[end] .+ colwidths[1:end-1]
-    coloverflows = coltextwidths .> displaysize(io)[2]
-    chunkbounds = getchunkbounds(maxwidths, splitcols || !allcols, displaysize(io)[2])
 
-    nchunks = length(chunkbounds) - 1
-    nchunks = allcols ? nchunks : (chunkbounds[2] < 1 ? 0 : min(nchunks, 1))
-    nchunks = !allcols && splitcols && any(coloverflows) ? 0 : nchunks
+    chunkedcols = getchunkedcols(maxwidths, splitcols || !allcols, displaysize(io)[2])
+    chunkwidths = [colwidths[end] + sum(colwidths[cols]) for cols=chunkedcols]
+    chunkoverflows = chunkwidths .> displaysize(io)[2]
+
+    if !allcols && any(chunkoverflows)
+        chunkedcols = chunkedcols[1:(findfirst(chunkoverflows) - 1)]
```

Comment:
why not this and omit the next condition?
```suggestion
        chunkedcols = chunkedcols[1:max(1, findfirst(chunkoverflows) - 1)]
```

Reply:
This won't cover the case where no chunk overflows the display, but because `allcols == false`, only the first chunk is to be displayed anyways. I've reworked the code a bit to make it a bit clearer and will include changes in an upcoming commit.

Reply:
sorry I have meant `min` not `max` - fixed above.

Reply:
No worries, I understood what you were getting at. This still misses the case that I mentioned above.

If there's an alternative to `findfirst(...)` with a default value if nothing is found then we can clean this up quite a bit more, but as is we need to avoid `findfirst` returning `nothing` and throwing an error before we collapse the two conditionals.

Reply:
no matter if you changed it and it works. To avoid producing `nothing` use `something(findfirst(...), the_value_you_want_on_nothing)`